### PR TITLE
feat: region specific support in config [DM-1312]

### PR DESCRIPTION
**Pull request by jacobfrackson**
## Description
Locally a region may already be set, but Airflow requires that a region be explicitly set. Region is now supported as a config param.

## Related JIRA issues
[DM-1312]

## Checklist

- [x] PR title follows [Commit Convention](https://www.notion.so/godialogue/Commit-Convention-84fd9a4c149e48c998d760f1c9176df0) 
- [x] PR title ends with a JIRA issue ID  
- [x] All relevant PR sections are populated, irrelevant ones are removed 
- [ ] [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests) is opened for WIP changes 
- [x] PR review requested from groups not individuals 
- [ ] PR link is posted to the corresponding Slack channel 

[DM-1312]: https://dialoguemd.atlassian.net/browse/DM-1312

### CODE REVIEW

File: dia_tap_xero/credentials.py [hunk 1]
```diff
@@ -13,61 +13,66 @@
...
+                "Content-Type": "application/x-www-form-urlencoded",
+            },
+            data={
+                "grant_type": "refresh_token",
+                "refresh_token": self.config["refresh_token"],
+            },
+        )
+        json_response = response.json()
+
+        self.config = self.rotate_refresh_tokens(json_response)
```
Comment (mdimercurio) [inline dia_tap_xero/credentials.py]: is the assignment to `self.config` required here? `rotate_refresh_tokens` already updates the `refresh_token` and `access_token` so I think this is reassigning the same values

File: dia_tap_xero/credentials.py [hunk 2]
```diff
@@ -13,61 +13,66 @@
...
-    )
-    json_response = response.json()
-
-    config = rotate_refresh_tokens(config, json_response)
-
-    LOGGER.info("Credentials refreshed, new tokens saved to config")
-    return config
+class XeroCredentials(object):
+    def __init__(self, config):
+        self.SSM = boto3.client("ssm", region_name=config["region_name"])
```
Comment (mdimercurio) [inline dia_tap_xero/credentials.py:19]: can you update the README with region_name being required in the config and the default value that we use

**Review by mdimercurio — CHANGES_REQUESTED** [review]

File: dia_tap_xero/credentials.py (hunk 1 above)
Comment (jacobfrackson) [reply dia_tap_xero/credentials.py]: ah good point. this was old from before it was a class

File: dia_tap_xero/credentials.py (hunk 2 above)
Comment (jacobfrackson) [reply dia_tap_xero/credentials.py:19]: good point. also updated the required config fields in the init

**Review by mdimercurio — APPROVED** [review]

**Pull request merged**